src/components/LoadDatasetButton.tsx

```diff
@@ -304,16 +321,53 @@ export default function LoadDatasetButton(props: LoadDatasetButtonProps): ReactE
                     ref={inputRef}
                     onChange={(event) => setUrlInput(event.target.value)}
                     allowClear
-                    disabled={isLoading}
+                    disabled={isLoadingUrl || isLoadingZip}
                     onPressEnter={handleLoadClicked}
                   />
                 </Dropdown>
-                <Button type="primary" onClick={handleLoadClicked} loading={isLoading}>
+                <Button type="primary" onClick={handleLoadClicked} loading={isLoadingUrl} disabled={isLoadingZip}>
                   Load
                 </Button>
               </Space.Compact>
             </div>
           </div>
+
+          <FlexRowAlignCenter>
+            <Upload
+              name="file"
+              multiple={false}
+              accept=".zip"
+              showUploadList={false}
+              beforeUpload={async (zipFile: File): Promise<boolean> => {
+                if (isLoadingUrl || isLoadingZip) {
+                  return false;
+                }
+                setErrorText("");
+                setIsLoadingZip(true);
+                const fileMap = await zipToFileMap(zipFile);
+                if (Object.keys(fileMap).length === 0) {
+                  setErrorText("No files found in ZIP file.");
+                  return false;
```

Comment:
I think there should be a `setIsLoadingZip(false);` before returning here?

Reply:
Good catch, thank you!